.travis.yml

```diff
@@ -20,142 +20,122 @@
+   - export SOURCEDIR=$PWD
+   - echo $PWD
+   - ls $PWD
+   - cp music_install.sh libneurosim-csa_install.sh $HOME
+   - cd $HOME/build
+   - echo $PATH
+  # Upgrade pip and setuptools
+   - pip install -U pip
+   - pip install -U setuptools
+  
   # Install packages required by NEST.
   # (Please see also: http://www.nest-simulator.org/installation/#standard-configuration)
-  - sudo apt-get install -y build-essential cmake libltdl7-dev libreadline6-dev libncurses5-dev libgsl0-dev python-all-dev python-numpy python-scipy python-matplotlib ipython
-
-  # Install pip and Cython.
-  - wget https://bootstrap.pypa.io/get-pip.py
-  - sudo python get-pip.py
-  - sudo pip install -U setuptools
-  - pip install --user cython==0.23.4
-
-  # Install package terminaltables required by parse_travis_log.py to create the build summary.
-  - sudo pip install terminaltables
-
-  # Install OpenMPI.
-  - sudo apt-get install -y openmpi-bin libopenmpi-dev
-
-  # Install nose framework for the Python testsuite.
-  # (Please note that the nose-capturestderr plugin is needed to suppress NEST output to stderr which may interfere 
-  #  with the nosetests output hindering correct log-parsing. stdout it captured by default.)
-  - sudo apt-get install -y python-nose
-  - sudo pip install "nose-capturestderr==1.2"
-
-  # Install boost libraries required for VERA++ static code analysis.
-  - sudo apt-get install -y libboost-filesystem-dev libboost-program-options-dev libboost-regex-dev libboost-wave-dev libboost-python-dev
-
-  # Install pep8 for static code analysis.
-  - sudo apt-get install -y pep8
-
-  # Required for building cppcheck-1.69.
-  - sudo apt-get install -y libpcre3 libpcre3-dev
-
-  # Install jq to process JSON output from GitHub API.
-  - sudo apt-get install -y jq
-
-  # Install libneurosim
-  - git clone https://github.com/INCF/libneurosim.git
-  - pushd libneurosim
-  - ./autogen.sh
-  - ./configure --prefix=$HOME/.cache/libneurosim.install --with-mpi
-  - make
-  - make install
-  - popd
-
-  # Install csa
-  - git clone https://github.com/INCF/csa.git
-  - pushd csa
-  - ./autogen.sh
-  - ./configure --with-libneurosim=$HOME/.cache/libneurosim.install --prefix=$HOME/.cache/csa.install
-  - make
-  - make install
-  - popd
-
-  # Install MUSIC
-  - sudo apt-get install -y python-mpi4py
-  - git clone https://github.com/INCF/MUSIC.git music
-  - pushd music
-  - ./autogen.sh
-  - ./configure --prefix=$HOME/.cache/music.install
-  - make
-  - make install
-  - popd
-
-  # Change directory back to the NEST source code directory.
-  - cd $SOURCEDIR
-
+  # terminaltables is required by parse_travis_log.py to create the build summary.
+   - pip install cython scipy matplotlib terminaltables mpi4py
+  # Install boost libraries required for VERA++ static code analysis.   
+   - sudo apt-get install -y libboost-filesystem-dev libboost-program-options-dev libboost-regex-dev libboost-wave-dev libboost-python-dev
```

Comment:
What is the reason for installing these Python libraries via `pip` and the Ubuntu packages manually using `apt-get` instead of adding all of them to the `addons` block above? Please either move them up and/or comment here on the reason.

Reply:
Moved the boost library packages to the list of travis add ons source list which would save the bootstrapping time when compared to the installation from the ubuntu repo using apt-get. As few packages have an older version under the add ons list,  installing those via pip